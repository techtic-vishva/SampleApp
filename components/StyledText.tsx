import { Text, TextProps } from './Themed'

export function MonoText(props: TextProps) {
	return <Text {...props} style={[props.style, { fontFamily: 'space-mono' }]} />
}

export function MainText(props: TextProps) {
	return <Text {...props} style={[props.style, { fontFamily: 'objektiv' }]} />
}

export function HeaderText(props: TextProps) {
	return <Text {...props} style={[{ fontFamily: 'objektiv-semi-bold', fontSize: 20 }, props.style]} />
}
