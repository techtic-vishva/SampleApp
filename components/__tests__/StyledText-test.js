import * as React from 'react'
import renderer from 'react-test-renderer'

import { MonoText, HeaderText, MainText } from '../StyledText'

it(`MonoText renders correctly`, () => {
	const tree = renderer.create(<MonoText>Snapshot test!</MonoText>).toJSON()

	expect(tree).toMatchSnapshot()
})

it(`HeaderText renders correctly`, () => {
	const tree = renderer.create(<HeaderText>Snapshot test!</HeaderText>).toJSON()

	expect(tree).toMatchSnapshot()
})

it(`MainText renders correctly`, () => {
	const tree = renderer.create(<MainText>Snapshot test!</MainText>).toJSON()

	expect(tree).toMatchSnapshot()
})
