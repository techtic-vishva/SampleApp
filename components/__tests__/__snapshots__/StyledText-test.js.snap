// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#fff",
        "fontFamily": "objektiv",
      },
      Array [
        Object {
          "fontFamily": "objektiv-semi-bold",
          "fontSize": 20,
        },
        undefined,
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`MainText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#fff",
        "fontFamily": "objektiv",
      },
      Array [
        undefined,
        Object {
          "fontFamily": "objektiv",
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`MonoText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#fff",
        "fontFamily": "objektiv",
      },
      Array [
        undefined,
        Object {
          "fontFamily": "space-mono",
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;
